### ISSUE TITLE
rule eqeqeq without option on a == null does not warn

### ISSUE THREAD

**Issue by fediev** [eslint/eslint#3699]
eslint version: v1.3.1
rule eqeqeq with no option on a == null expects a warning, but no warning. 

```javascript 
/* eslint eqeqeq: 2 */
var x = 1;
x == null;
```

eslint/lib/rules/eqeqeq.js line 74 might be a bug. 
```javascript 
            if (context.options[0] === "smart" && (isTypeOfBinary(node) ||
-                    areLiteralsAndSameType(node)) || isNullCheck(node)) {
+                    areLiteralsAndSameType(node) || isNullCheck(node))) {
                return;
            }
```

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by IanVS** [comment]
Verified in demo.  Since it looks like you've already found the bug, would you like to submit a PR, @fediev?

**Comment by fediev** [comment]
OK, I will make a PR.

**Comment by IanVS** [comment]
Thanks! If you haven't submitted a PR to ESLint before, please take a read through the [contributing guide](http://eslint.org/docs/developer-guide/contributing/).

**Comment by mysticatea** [comment]
Thank you!
